include/circt/Dialect/Ibis/IbisOps.td

```diff
@@ -203,61 +269,40 @@ def OutputPortOp : PortLikeOp<"port.output"> {
   let summary = "Ibis output port";
 }
 
+class PortRefToInnerTypeConstraint<string lhs, string rhs>
+  : TypesMatchWith<"the lhs portref dictates the rhs type",
+    lhs, rhs, "$_self.cast<PortRefType>().getPortType()">;
```

Comment:
I'm kinda surprised the cast is necessary.

Reply:
`$_self` generates as `*this->getODSOperands(0).begin()).getType()`